### PR TITLE
feat: Basic vQFX QCOW/VMDK and Containerlab Support

### PR DESCRIPTION
**Pull request by chriscummings-esnet**
This adds very basic support for vQFX in containerlab. This combines the work in https://github.com/hellt/vrnetlab/pull/49 and https://github.com/vrnetlab/vrnetlab/pull/234 as well as making it hopefully easy enough for a user to build multiple types of vQFX (older VMDK based or newer qcow based images).  This works in conjunction with https://github.com/srl-labs/containerlab/pull/626 and seems to work just fine in my testing so far (vQFX 20.2R1.10 and 18.4R1.8 have been tested so far, but if someone wants to test with other images I'd be glad to help, I just don't have access to any other images currently.)

I still need to update docs, but if the code looks good and ready to merge, let me know and I can update them before we merge. Unfortunately I couldn't find a good way to auto-convert the vmdks to qcows using the default make target without re-writing the make includes, so I just had to add a separate target that doesn't work as a dependency (long story short, it has to do with the way our make includes happen) and has to be manually run. For now, to build a vmdk image, you have to first run `make format-legacy-images` and then run the normal `make`. 

Let me know what you think!

### CODE REVIEW

**Comment by hellt** [comment]
I wonder can this be simplified the same way as was done for vmx https://github.com/hellt/vrnetlab/blob/master/vmx/Makefile?

**Comment by chriscummings-esnet** [comment]
Unfortunately I can't find a good way to make it that simple, but I'm very open to ideas. Essentially what happens is that the $(IMAGES) variable is defined right off the bat in the makefile includes, and that really needs to be defined _after_ the old-style vmdk images are converted to .qcow. And since the include doesn't define $(IMAGES) inside of a target, we can't really over-ride how it's defined just for vQFX, so it would require changing that in the main makefile, which I fear would have some nasty knock-on effects for other vrnetlab make files that depend on it.

File: vqfx/Makefile [hunk 1]
```diff
@@ -1,16 +1,35 @@
...
 
-# vqfx10k-pfe-20160609-2.vmdk
+format-legacy-images:
+	@if ls *.vmdk; then echo "VMDKs exist, converting them to qcow format"; qemu-img convert -f vmdk -O qcow2 *-re-*.vmdk vqfx-$(VMDK_VERSION)-re-qemu.qcow2 && qemu-img convert -f vmdk -O qcow *-pfe-*.vmdk vqfx-$(VMDK_VERSION)-pfe-qemu.qcow; echo "VMDKs have been converted"; fi
+
 # TODO: we should make sure we only copy one PFE image (the latest?), in case there are many
 docker-pre-build:
-	cp vqfx10k-pfe*.vmdk docker/
+
+	echo "pfe     $(PFE_IMAGE)"
```
Comment (hellt) [inline vqfx/Makefile:23]: why do we copy only the pfe image? what happens with re?

Comment (chriscummings-esnet) [reply vqfx/Makefile:23]: I am not entirely sure as that came from the PR#234 in the upstream repo, however, I _think_ the RE gets copied in and added to the image at [line 9](https://github.com/hellt/vrnetlab/blob/7542cf1cceb266935a3f59dbbc2bc63dcae1639c/makefile.include#L9) of the `makefile.include` or possibly during [line 34](https://github.com/hellt/vrnetlab/blob/7542cf1cceb266935a3f59dbbc2bc63dcae1639c/vqfx/Makefile#L34) of this Makefile, which calls the make target on [line 21](https://github.com/hellt/vrnetlab/blob/7542cf1cceb266935a3f59dbbc2bc63dcae1639c/makefile.include#L21) of the `makefile.include`.  This entire step of copying the PFE might actually be redundant and not needed. Let me test without it and see what happens.

Comment (chriscummings-esnet) [reply vqfx/Makefile:23]: Okay, so it is not redundant and is definitely needed. Without having dug too deep, I'm guessing that the built-in stuff only handles 1 disk image, so this is necessary to get the PFE into docker.

**Comment by hellt** [comment]
@chriscummings-esnet let me merge this as an experimental stuff
I wiped out the readme so that we can add it later on when we know the tested version and the `make` process

**Comment by chriscummings-esnet** [comment]
Okay, fantastic!

**Pull request merged**